gateway/session_manager.go

```diff
@@ -34,8 +38,34 @@ const (
 // SessionLimiter is the rate limiter for the API, use ForwardMessage() to
 // check if a message should pass through or not
 type SessionLimiter struct {
-	bucketStore leakybucket.Storage
-	Gw          *Gateway `json:"-"`
+	drlManager     *drl.DRL
+	config         *config.Config
+	bucketStore    leakybucket.Storage
+	limiterStorage redis.UniversalClient
+}
+
+// NewSessionLimiter initializes the session limiter.
+//
+// The session limiter initializes the storage required for rate limiters.
+// It supports two storage types: `redis` and `local`. If redis storage is
+// configured, then redis will be used. If local storage is configured, then
+// in-memory counters will be used. If no storage is configured, it falls
+// back onto the default gateway storage configuration.
+func NewSessionLimiter(conf *config.Config, drlManager *drl.DRL) SessionLimiter {
+	sessionLimiter := SessionLimiter{
+		drlManager:  drlManager,
+		config:      conf,
+		bucketStore: memorycache.New(),
+	}
+
+	storageConf := conf.GetRateLimiterStorage()
+
+	switch storageConf.Type {
+	case "redis":
```

Comment:
why use switch and not just a conditional?

Reply:
The value should have a known set of valid storage types. In time, we want this to be covered with [exhaustive](https://pkg.go.dev/github.com/nishanths/exhaustive) so that we know we handle all known storage types.